### PR TITLE
Feat(eos_designs): Add wan_carrier and circuit_id to l3 interface description

### LINKED ISSUE #3527: Feat(eos_designs): Update the description of WAN interfaces to include WAN carrier and circuit ID
### Enhancement summary

For `nodes.l3_interfaces` where the WAN keys are defined (`wan_carrier` and `wan_circuit_id`), include them in the interface description

### Which component of AVD is impacted

eos_designs

### Use case example

Better info in the config 

### Describe the solution you would like

Move the description to the https://github.com/aristanetworks/avd/blob/devel/ansible_collections/arista/avd/roles/eos_designs/python_modules/interface_descriptions/avdinterfacedescriptions.py file and call it from the module

Add the WAN carrier and WAN circuit ID when they exist

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by ayushmittal-arista**
## Change Summary

Add wan_carrier and circuit_id to l3 interface description

## Related Issue(s)

Fixes #3527

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes
.

## How to test
molecule

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions.py
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions.py]: we cannot change the signature of this function. Please revert these changes.

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/interface_descriptions/avdinterfacedescriptions.py
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/interface_descriptions/avdinterfacedescriptions.py]: This signature should not be changed, since it is part of a class we expose for customization of descriptions. Please revert these changes. See the docstring on the class.

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/interface_descriptions/avdinterfacedescriptions.py
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/interface_descriptions/avdinterfacedescriptions.py:49]: Instead of modifying the legacy function below, you can detect wan_carrier and/or wan_circuit_id here and just append it to the generated description from the other function. This is needed in case of this being sub-classed by a customer.

Comment (ayushmittal-arista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/interface_descriptions/avdinterfacedescriptions.py:49]: Yea, got it.

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/structured_configs/node-type-l3-interfaces.yml
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/structured_configs/node-type-l3-interfaces.yml]: This should not be removed. It is probably caused by the changes to the custom modules. Once they are reverted, this should come back.

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/interface_descriptions/avdinterfacedescriptions.py
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/interface_descriptions/avdinterfacedescriptions.py]:
```suggestion
        if data.wan_carrier or data.wan_circuit_id:
```

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions_with_data.py
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions_with_data.py:1]: Please revert all changes from this file.

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions.py
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions.py:1]: Please revert all changes from this file.

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/interface_descriptions/avdinterfacedescriptions.py
Comment (ayushmittal-arista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/interface_descriptions/avdinterfacedescriptions.py]: can wan circuit id be present without wan carrier ?

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/interface_descriptions/avdinterfacedescriptions.py]: Well there is nothing in the logic before this step that hinders that, so we should not infer such a restriction this deep in the chain.

Comment (ayushmittal-arista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/interface_descriptions/avdinterfacedescriptions.py]: Comment no longer applicable, changed code to fix the bug you pointed. 

The bug was not due to custom interface descriptions module, but rather due to `if data.wan_carrier` check, as for l3_interfaces that didn't have wan_carrier we were not emitting anything.

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions.py
Comment (ayushmittal-arista) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions.py:1]: Reverted

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions_with_data.py
Comment (ayushmittal-arista) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions_with_data.py:1]: The bug was not due to these changes. Should this still be reverted

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions_with_data.py:1]: Yes. Thanks

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/structured_configs/autovpn-edge-no-default-policy.yml
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/structured_configs/autovpn-edge-no-default-policy.yml:119]: Could we see a description where there is the peer_ip / peer_name ?

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions_with_data.py
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions_with_data.py]: not sure if this is really needed here but thats ok

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions_with_data.py]: I have asked for these changes to be removed.

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/structured_configs/autovpn-edge-no-default-policy.yml
Comment (ayushmittal-arista) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/structured_configs/autovpn-edge-no-default-policy.yml:119]: Done.

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions_with_data.py
Comment (ayushmittal-arista) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/custom_modules/custom_interface_descriptions_with_data.py]: Remvoed.

**Review by ClausHolbechArista — APPROVED** [review]

**Comment by ClausHolbechArista** [comment]
This one needs a rerun of the molecule schenario `eos_designs_unit_tests`. It can wait until we have two approvals since it will be needed everytime something wan related is merged.

**Review by gmuloc — APPROVED** [review]
LGTM - lets rerun molecule and get it merged

**Pull request merged**
